gcloud-java-datastore/src/main/java/com/google/gcloud/datastore/DatastoreFactory.java

```diff
@@ -16,18 +16,29 @@
 
 package com.google.gcloud.datastore;
 
+import com.google.gcloud.ServiceFactory;
+import java.io.ObjectStreamException;
 
 /**
  * A base class for Datastore factories.
  */
-public abstract class DatastoreFactory {
+public abstract class DatastoreFactory implements ServiceFactory<Datastore, DatastoreOptions> {
+
+  private static final long serialVersionUID = 5037190305022535983L;
 
   private static final DatastoreFactory INSTANCE = new DatastoreFactory() {
```

Comment:
Since the actual class is part of the serial form, you can't leave this as an anonymous class if you are making it serializable. Otherwise the serial form would depend on the name `DatastoreFactory$1` which is obviously much too fragile. The most straightforward fix is this:
```java
  private static final class Impl extends DatastoreFactory {
    private static final long serialVersionUID = ...
    ...
  }
  private static final DatastoreFactory INSTANCE = new Impl();
```

I'm not sure I understand whether it's worth distinguishing DatastoreFactory.Impl from plain DatastoreFactory, though.